### PR TITLE
Release 2.3.0

### PR DESCRIPTION
**Pull request by jonasscheid**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/epitopeprediction _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by christopher-mohr** [comment]
I just removed the "old" CI tests from the required checks

File: assets/schema_input.json [hunk 1]
```diff
@@ -20,20 +20,32 @@
...
                 "pattern": "^(I|II|H-2)$",
-                "errorMessage": "The MHC class must be provided. Valid values: "
+                "errorMessage": "The MHC class must be provided. Valid values: 'I', 'II' or 'H-2'"
             },
             "filename": {
                 "type": "string",
-                "pattern": "^\\S+\\.(vcf|tsv|fasta|fa|txt)$",
-                "errorMessage": "Variants/proteins/peptides for sample must be provided and have one of the following extensions:  '.vcf', '.tsv', '.fasta', '.fa', '.txt'"
+                "pattern": "^\\S+\\.(vcf|vcf.gz|tsv|fasta|fa)$",
+                "errorMessage": "Variants/proteins/peptides for sample must be provided and have one of the following extensions: '.vcf', '.vcf.gz', '.tsv', '.fasta', '.fa'"
```
Comment (christopher-mohr) [inline assets/schema_input.json:48]:
```suggestion
                "errorMessage": "Variants/proteins/peptides of a sample must be provided and have one of the following extensions: '.vcf', '.vcf.gz', '.tsv', '.fasta', '.fa'"
```

File: docs/usage.md [hunk 2]
```diff
@@ -16,12 +16,16 @@ You will need to create a samplesheet with information about the samples you wou
...
-
-:warning: Please note that genomic variants have to be annotated. Currently, we support variants that have been annotated using [SnpEff](http://pcingola.github.io/SnpEff/). Support for [VEP](https://www.ensembl.org/info/docs/tools/vep/index.html) will be available with one of the upcoming versions.
+The pipeline currently accepts three different types of input that are genomic variants, peptides and proteins.
 
 #### Genomic variants
 
+The supported file formats for genomic variants are `.vcf`, `.vcf.gz` and `tsv`.
+
+> [!IMPORTANT]
+> Please note that genomic variants have to be annotated. Currently, we support variants that have been annotated using [SnpEff](http://pcingola.> github.io/SnpEff/).
```
Comment (christopher-mohr) [inline docs/usage.md:26]:
```suggestion
> Please note that genomic variants have to be annotated. Currently, we support variants that have been annotated using [SnpEff](http://pcingola.> github.io/SnpEff/) and [VEP](https://www.ensembl.org/info/docs/tools/vep/index.html).
```

File: docs/usage.md [hunk 3]
```diff
@@ -16,12 +16,16 @@ You will need to create a samplesheet with information about the samples you wou
...
+> Support for [VEP](https://www.ensembl.org/info/docs/tools/vep/index.html) will be available with one of the upcoming versions.
```
Comment (christopher-mohr) [inline docs/usage.md:27]:
```suggestion
```

File: docs/usage.md [hunk 4]
```diff
@@ -32,6 +36,13 @@ chr1 12954870 12954870 C T . 0 NORMAL:414,TUMOR:8 . missense_variant 0.5 transcr
 
 ```
 
+For genomic variants, reference information from `Ensembl BioMart` is used. The default database version is the most recent `GRCh37` version. If you want to do the predictions based on `GRCh38` as the reference genome, please specify `--genome_reference grch37` in your pipeline call.
```
Comment (christopher-mohr) [inline docs/usage.md:39]:
```suggestion
For genomic variants, reference information from `Ensembl BioMart` is used. The default database version is the most recent `GRCh37` version. If you want to do the predictions based on `GRCh38` as the reference genome, please specify `--genome_reference grch38` in your pipeline call.
```

File: lib/WorkflowEpitopeprediction.groovy [hunk 5]
```diff
@@ -39,32 +43,76 @@ class WorkflowEpitopeprediction {
...
+    //
+
+    public static String toolCitationText(params) {
+
+        // TODO nf-core: Optionally add in-text citation tools to this list.
+        // Can use ternary operators to dynamically construct based conditions, e.g. params["run_xyz"] ? "Tool (Foo et al. 2023)" : "",
+        // Uncomment function in methodsDescriptionText to render in MultiQC report
+        def citation_text = [
+                "Tools used in the workflow included:",
+                "FastQC (Andrews 2010),",
```
Comment (christopher-mohr) [inline lib/WorkflowEpitopeprediction.groovy:57]:
```suggestion
```

**Review by christopher-mohr — COMMENTED** [review]
A couple of suggestions so far, I will continue later.

File: bin/check_requested_models.py [hunk 6]
```diff
@@ -74,7 +74,12 @@ def __main__():
         }
 
     # get the alleles
-    alleles = [Allele(a) for a in args.alleles.split(";")]
+    if args.alleles.startswith("http"):
```
Comment (christopher-mohr) [inline bin/check_requested_models.py:77]: Not needed in this context since input should be string only.

File: bin/check_requested_models.py [hunk 7]
```diff
@@ -74,7 +74,12 @@ def __main__():
...
+        alleles = [Allele(a) for a in urllib.request.urlopen(args.alleles).read().decode("utf-8").splitlines()]
+    elif args.alleles.endswith(".txt"):
+        alleles = [Allele(a) for a in open(args.alleles).read().splitlines()]
+    else:
+        alleles = [Allele(a) for a in args.alleles.split(";")]
```
Comment (christopher-mohr) [inline bin/check_requested_models.py:82]:
```suggestion
alleles = [Allele(a) for a in args.alleles.split(";")]
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,26 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.3.0 - Oesterberg - 2024-02-14
+
+### `Changed`
+
+- [#228](https://github.com/nf-core/epitopeprediction/pull/228) - Update to nf-core template `2.12`
+- [#227](https://github.com/nf-core/epitopeprediction/pull/227) Prevent crash if no transcript is found (in splitted vcf)
```
Comment (christopher-mohr) [inline CHANGELOG.md:11]:
```suggestion
- [#227](https://github.com/nf-core/epitopeprediction/pull/227) - Prevent crash if no transcript is found (in splitted vcf)
```

**Review by christopher-mohr — COMMENTED** [review]
Some additional comments/suggestions

File: .nf-core.yml
[the remark this answers is not in the document]
Comment (jonasscheid) [reply .nf-core.yml:1]: I tried this as well, but did not work here somehow. Will be handled already differently in the upcoming tools version afaik, so agree to skip it 👍🏼

File: docs/usage.md [hunk 9]
```diff
@@ -16,12 +16,15 @@ You will need to create a samplesheet with information about the samples you wou
...
 ### Input Formats
 
-The pipeline currently accepts three different types of input that are genomic variants, peptides and proteins. The supported file formats for genomic variants are `.vcf`, `.vcf.gz` and `tsv`.
-
-:warning: Please note that genomic variants have to be annotated. Currently, we support variants that have been annotated using [SnpEff](http://pcingola.github.io/SnpEff/). Support for [VEP](https://www.ensembl.org/info/docs/tools/vep/index.html) will be available with one of the upcoming versions.
+The pipeline currently accepts three different types of input that are genomic variants, peptides and proteins.
 
 #### Genomic variants
 
+The supported file formats for genomic variants are `.vcf`, `.vcf.gz` and `tsv`.
```
Comment (mapo9) [inline docs/usage.md:23]: remove `tsv` here.

File: docs/usage.md [hunk 10]
```diff
@@ -124,17 +157,26 @@ First, go to the [nf-core/epitopeprediction releases page](https://github.com/nf
 
 This version number will be logged in reports when you run the pipeline, so that you'll know what you used when you look back in the future. For example, at the bottom of the MultiQC reports.
 
+To further assist in reproducbility, you can use share and re-use [parameter files](#running-the-pipeline) to repeat pipeline runs with the same settings without having to write out a command with every single parameter.
```
Comment (mapo9) [inline docs/usage.md:160]:
```suggestion
To further assist in reproducibility, you can use share and re-use [parameter files](#running-the-pipeline) to repeat pipeline runs with the same settings without having to write out a command with every single parameter.
```

File: docs/usage.md [hunk 11]
```diff
@@ -124,17 +157,26 @@ First, go to the [nf-core/epitopeprediction releases page](https://github.com/nf
...
+
+:::tip
+If you wish to share such profile (such as upload as supplementary material for academic publications), make sure to NOT include cluster specific paths to files, nor institutional specific profiles.
```
Comment (mapo9) [inline docs/usage.md:163]:
```suggestion
If you wish to share such profile (such as upload as supplementary material for academic publications), make sure NOT to include cluster specific paths to files, nor institution specific profiles.
```

File: nextflow.config [hunk 12]
```diff
@@ -9,15 +9,17 @@
...
-    peptides_split_minchunksize = 5000
-    split_by_variants = false
-    split_by_variants_size = 0
-    split_by_variants_distance = 110000
+    input                        = null
+    peptides_split_maxchunks     = 100
+    peptides_split_minchunksize  = 5000
+    split_by_variants            = false
+    split_by_variants_size       = 0
+    split_by_variants_distance   = 110000
```
Comment (mapo9) [inline nextflow.config:18]: I think best practice would be to have all the `=`at the same tab position?

File: nextflow_schema.json [hunk 13]
```diff
@@ -68,6 +68,22 @@
             "fa_icon": "fas fa-dna",
             "description": "Reference genome related files and options required for the workflow.",
             "properties": {
+                "genome": {
+                    "type": "string",
+                    "description": "Name of iGenomes reference.",
+                    "fa_icon": "fas fa-book",
+                    "help_text": "If using a reference genome configured in the pipeline using iGenomes, use this parameter to give the ID for the reference. This is then used to build the full paths for all required reference genome files e.g. `--genome GRCh38`. \n\nSee the [nf-core website docs](https://nf-co.re/usage/reference_genomes) for more details."
```
Comment (mapo9) [inline nextflow_schema.json:75]:
```suggestion
                    "help_text": "If using a reference genome configured in the pipeline using iGenomes, use this parameter to give the ID for the reference. This is then used to build the full paths for all required reference genome files e.g. `--genome GRCh38`.\n\nSee the [nf-core website docs](https://nf-co.re/usage/reference_genomes) for more details."
```

File: nextflow_schema.json [hunk 14]
```diff
@@ -68,6 +68,22 @@
...
+                },
+                "fasta": {
```
Comment (mapo9) [inline nextflow_schema.json:77]: dont think you guys use this?

**Review by mapo9 — CHANGES_REQUESTED** [review]
looks very nice!
Only some minor stuff

**Comment by christopher-mohr** [comment]
Let's get another review for this because once tests pass because of the most recent template change.

**Review by mapo9 — APPROVED** [review]
LGTM! 🚀 

just one minor change to improve readability of a parameter

**Review by christopher-mohr — APPROVED** [review]
LGTM

**Pull request merged**